docs/sharding.md

```diff
@@ -6,18 +6,23 @@ menu: thanos
 
 # Background
 
-Currently all components that read from object store assume that all the operations and functionality should be done based
+Currently, all components that read from object store assume that all the operations and functionality should be done based
 on **all** the available blocks that are present in the certain bucket's root directory.
 
 This is in most cases totally fine, however with time and allowance of storing blocks from multiple `Sources` into the same bucket,
 the number of objects in a bucket can grow drastically.
 
 This means that with time you might want to scale out certain components e.g:
 
-* Compactor: Larger number of objects does not matter much, however compactor has to scale (CPU, network, disk, memory) with number of Sources pushing blocks to the object storage.
+## Compactor
 
-* Store Gateway: Queries against store gateway which are touching large number of Sources might be expensive, so it has to scale up with number of Sources if we assume those queries.
-    * Orthogonally we did not advertise any labels on Store Gateway's Info. This means that querier was not able to do any pre-filtering, so all store gateways in system are always touched for each query.
+Larger number of objects does not matter much, however compactor has to scale (CPU, network, disk, memory) with number of Sources pushing blocks to the object storage.
+
+## Store Gateway
+
+Queries against store gateway which are touching large number of Sources might be expensive, so it has to scale up with number of Sources if we assume those queries.
+
+* Orthogonally we did not advertise any labels on Store Gateway's Info. This means that querier was not able to do any pre-filtering, so all store gateways in system are always touched for each query.
```

Comment:
But that's not true for a long time, right? We advertise all available pairs of external labels on StoreGW.

Reply:
Obsolete part, good point.